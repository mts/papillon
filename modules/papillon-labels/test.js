import React from 'react';
import {
  labelDefault,
  design,
  labelThemeGray,
  labelThemeDarkGray,
  labelThemeOrange,
  labelOutline,
  labelOutlineGreen,
  state,
  stateThemes,
} from './flavor';

describe('<Label />', () => {
  let wrapper;

  describe('Instance', () => {
    test('must be an instance of Label', () => {
      wrapper = renderMount(labelDefault);
      expect(wrapper.find('Label')).toHaveLength(1);
    });
  });

  describe('Snaphot', () => {
    test('must match labelDefault', () => {
      expect(global.renderToJSON(labelDefault)).toMatchSnapshot();
    });

    test('must match design', () => {
      expect(global.renderToJSON(design)).toMatchSnapshot();
    });

    test('must match labelThemeGray', () => {
      expect(global.renderToJSON(labelThemeGray)).toMatchSnapshot();
    });

    test('must match labelThemeDarkGray', () => {
      expect(global.renderToJSON(labelThemeDarkGray)).toMatchSnapshot();
    });

    test('must match labelThemeOrange', () => {
      expect(global.renderToJSON(labelThemeOrange)).toMatchSnapshot();
    });

    test('must match labelOutline', () => {
      expect(global.renderToJSON(labelOutline)).toMatchSnapshot();
    });

    test('must match labelOutlineGreen', () => {
      expect(global.renderToJSON(labelOutlineGreen)).toMatchSnapshot();
    });

    test('must match state', () => {
      expect(global.renderToJSON(state)).toMatchSnapshot();
    });

    test('must match stateThemes', () => {
      expect(global.renderToJSON(stateThemes)).toMatchSnapshot();
    });
  });
});
