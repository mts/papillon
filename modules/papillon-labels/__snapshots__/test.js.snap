// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> Snaphot must match design 1`] = `
<span
  className="Label bg-blue ml-1"
>
   
  design label
</span>
`;

exports[`<Label /> Snaphot must match labelDefault 1`] = `
<span
  className="Label bg-blue"
>
   
  default label
</span>
`;

exports[`<Label /> Snaphot must match labelThemeDarkGray 1`] = `
<span
  className="Label Label--gray-darker"
>
   
  dark gray label
</span>
`;

exports[`<Label /> Snaphot must match labelThemeGray 1`] = `
<span
  className="Label Label--gray"
>
   
  gray label
</span>
`;
